### PR TITLE
Add ECC hardware acceleration for TI platform

### LINKED ISSUE #11678: [cc13x2x7_26x2x7] TE#7 test harness timeouts
#### Problem

Matter test harness times out on automated tests for the CC13X2 / CC26X2 platform. This is due to the long processing time to complete PASE and CASE. While the sessions are established, the gross timeout of each test case fails the test run.

 #### Proposed Solution

Enable hardware acceleration for ECC operations on the TI platform.

### PR DESCRIPTION
**Pull request by srickardti**
#### Problem

* Fixes #11678

#### Change overview

Add MbedTLS ALT source files for the Texas Instruments CC13X2_26X2 platform to
enable hardware acceleration of ECDSA and ECDHE. This was done to address
timeout issues with PASE and CASE test cases in TE#6. The test cases would fail
due to a gross timeout even though the sessions were established. This change
requires a new version of the SimpleLink SDK to avoid a BLE ROM symbol
collision.

A new version of the SimpleLink SDK based on the 5.30 SDK is used here.
Documentation has been updated in associated project READMEs. The button
middleware driver API has been updated and associated calls in examples have
been updated to match.

Address an issue where the SHA-256 hardware acceleration did not copy the
digest buffer. It was incorrectly assumed that the digest buffer would be empty
when a SHA context was cloned. This addresses an issue in decoding ZCL
messages.

#### Testing

Unit testing with TE#6 test harness.

### CODE REVIEW

**Comment by woody-apple** [comment]
Fast tracking, given platform maintainer making platform changes.

File: examples/pump-app/cc13x2x7_26x2x7/README.md [hunk 1]
```diff
@@ -113,21 +113,21 @@ Ninja to build the executable.
...
-    `C:\ti`
+    `/home/username/ti/simplelink_cc13xx_cc26xx_sdk_5_30_03_01_eng` and
+    `/home/username/ti/sysconfig_1.10.0`. On Windows the default directory is
+    `C:\ti`.
 
     ```
-    $ cd ~/connectedhomeip/examples/pump-app/cc13x2_26x2
-    $ export TI_SIMPLELINK_SDK_ROOT=<path-to-simplelink-sdk>
-    $ export TI_SYSCONFIG_ROOT=<path-to-sysconfig-sdk>
+    $ cd ~/connectedhomeip/examples/lock-app/cc13x2x7_26x2x7
```
Comment (andersbangGF) [inline examples/pump-app/cc13x2x7_26x2x7/README.md:128]: lock-app -> pump app

**Review by andersbangGF — APPROVED** [review]

**Comment by srickardti** [comment]
I'm not sure what keeps erroring out the Darwin and QEMU builds. The change-set only touches TI device related files.

**Comment by srickardti** [comment]
@woody-apple , I'm not able to restart jobs. This PR only touches my platform files, I'm not sure why ESP32 keeps failing.

**Pull request merged**